### ISSUE TITLE
infinite loop when reaching EOF with some video formats like Macromedia Flash Video

### ISSUE THREAD

**Issue by Safari77** [mpv-player/mpv#5026]
### mpv version and platform
mpv 0.27.0-431-g265bdc7061 (C) 2000-2017 mpv/MPlayer/mplayer2 projects
 built on Sun Oct 22 15:24:10 EEST 2017
ffmpeg library versions:
   libavutil       56.0.100
   libavcodec      58.0.100
   libavformat     58.0.100
   libswscale      5.0.100
   libavfilter     7.0.100
   libswresample   3.0.100
ffmpeg version: N-115551-g64bc1a6516

Fedora 26 x86_64.

I don't know is the bug in ffmpeg or mpv, but ffmpeg has recent changes like
858db4b01fa2b55ee55056c033054ca54ac9b0fd d4d2e9fe4e1e93d37fd543092e45041f58fa8c00

### Reproduction steps
play flv till the end (you can also seek to the end)

### Expected behavior
mpv quits after playing the video

### Actual behavior
mpv stalls at the end, or does not start playing at all if video is short

### Log file
```
[   2.747][v][lavf] queuing seek to 4160.080000 (cascade)
```

```
[pid 25523] 15:30:33.356119 lseek(10, 1093744092, SEEK_SET) = 1093744092 <0.000008>
[pid 25523] 15:30:33.356144 read(10, "", 32768) = 0 <0.000008>
[pid 25523] 15:30:33.356170 lseek(10, 0, SEEK_END) = 1093744092 <0.000008>
[pid 25523] 15:30:33.356195 lseek(10, 1093744092, SEEK_SET) = 1093744092 <0.000007>
[pid 25523] 15:30:33.356223 read(10, "", 32768) = 0 <0.000009>
[pid 25523] 15:30:33.356250 lseek(10, 0, SEEK_END) = 1093744092 <0.000008>
[pid 25523] 15:30:33.356275 lseek(10, 1093744092, SEEK_SET) = 1093744092 <0.000008>
```
```
Thread 15 (Thread 0x7fb883fff700 (LWP 24943)):
#0  0x00007fb8c3d8c90b in futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x5601db3d6ec8) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x5601db3d6e78, cond=0x5601db3d6ea0) at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=cond@entry=0x5601db3d6ea0, mutex=mutex@entry=0x5601db3d6e78) at pthread_cond_wait.c:655
#3  0x00005601d91bb3bc in playthread (arg=0x5601db340ee0) at ../audio/out/push.c:397
#4  0x00007fb8c3d8636d in start_thread (arg=0x7fb883fff700) at pthread_create.c:456
#5  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 14 (Thread 0x7fb89cfa0700 (LWP 24942)):
#0  0x00007fb8c3ab2acd in poll () at ../sysdeps/unix/syscall-template.S:84
#1  0x00007fb8c9aa1c01 in poll_func () from target:/lib64/libpulse.so.0
#2  0x00007fb8c9a935c0 in pa_mainloop_poll () from target:/lib64/libpulse.so.0
#3  0x00007fb8c9a93c50 in pa_mainloop_iterate () from target:/lib64/libpulse.so.0
#4  0x00007fb8c9a93ce0 in pa_mainloop_run () from target:/lib64/libpulse.so.0
#5  0x00007fb8c9aa1b49 in thread () from target:/lib64/libpulse.so.0
#6  0x00007fb8c04373a8 in internal_thread_func () from target:/usr/lib64/pulseaudio/libpulsecommon-11.1.so
#7  0x00007fb8c3d8636d in start_thread (arg=0x7fb89cfa0700) at pthread_create.c:456
#8  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 13 (Thread 0x7fb89d9ac700 (LWP 24941)):
#0  0x00007fb8c3d8c90b in futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x5601db289760) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x5601db2897c8, cond=0x5601db289738) at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=cond@entry=0x5601db289738, mutex=mutex@entry=0x5601db2897c8) at pthread_cond_wait.c:655
#3  0x00007fb8ca4b8c4e in frame_worker_thread (arg=0x5601db289720) at libavcodec/pthread_frame.c:177
#4  0x00007fb8c3d8636d in start_thread (arg=0x7fb89d9ac700) at pthread_create.c:456
#5  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 12 (Thread 0x7fb89e1ad700 (LWP 24940)):
#0  0x00007fb8c3d8c90b in futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x5601db2895b0) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x5601db289618, cond=0x5601db289588) at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=cond@entry=0x5601db289588, mutex=mutex@entry=0x5601db289618) at pthread_cond_wait.c:655
#3  0x00007fb8ca4b8c4e in frame_worker_thread (arg=0x5601db289570) at libavcodec/pthread_frame.c:177
#4  0x00007fb8c3d8636d in start_thread (arg=0x7fb89e1ad700) at pthread_create.c:456
#5  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 11 (Thread 0x7fb89e9ae700 (LWP 24939)):
#0  0x00007fb8c3d8c90b in futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x5601db289404) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x5601db289468, cond=0x5601db2893d8) at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=cond@entry=0x5601db2893d8, mutex=mutex@entry=0x5601db289468) at pthread_cond_wait.c:655
#3  0x00007fb8ca4b8c4e in frame_worker_thread (arg=0x5601db2893c0) at libavcodec/pthread_frame.c:177
#4  0x00007fb8c3d8636d in start_thread (arg=0x7fb89e9ae700) at pthread_create.c:456
#5  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 10 (Thread 0x7fb89f1af700 (LWP 24938)):
#0  0x00007fb8c3d8c90b in futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x5601db289250) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x5601db2892b8, cond=0x5601db289228) at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=cond@entry=0x5601db289228, mutex=mutex@entry=0x5601db2892b8) at pthread_cond_wait.c:655
#3  0x00007fb8ca4b8c4e in frame_worker_thread (arg=0x5601db289210) at libavcodec/pthread_frame.c:177
#4  0x00007fb8c3d8636d in start_thread (arg=0x7fb89f1af700) at pthread_create.c:456
#5  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 9 (Thread 0x7fb89f9b0700 (LWP 24937)):
#0  0x00007fb8c3d8c90b in futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x5601db2890a0) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x5601db289108, cond=0x5601db289078) at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=cond@entry=0x5601db289078, mutex=mutex@entry=0x5601db289108) at pthread_cond_wait.c:655
#3  0x00007fb8ca4b8c4e in frame_worker_thread (arg=0x5601db289060) at libavcodec/pthread_frame.c:177
#4  0x00007fb8c3d8636d in start_thread (arg=0x7fb89f9b0700) at pthread_create.c:456
#5  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 8 (Thread 0x7fb8a6c7e700 (LWP 24936)):
#0  0x00007fb8c3d8f666 in futex_abstimed_wait_cancelable (private=0, abstime=0x0, expected=0, futex_word=0x7fb8a04822f8) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
#1  do_futex_wait (sem=sem@entry=0x7fb8a04822f8, abstime=0x0) at sem_waitcommon.c:111
#2  0x00007fb8c3d8f758 in __new_sem_wait_slow (sem=sem@entry=0x7fb8a04822f8, abstime=0x0) at sem_waitcommon.c:181
#3  0x00007fb8c3d8f7c9 in __new_sem_wait (sem=sem@entry=0x7fb8a04822f8) at sem_wait.c:42
#4  0x00005601d927dc38 in screensaver_thread (arg=0x7fb8a0481d60) at ../video/out/x11_common.c:509
#5  0x00007fb8c3d8636d in start_thread (arg=0x7fb8a6c7e700) at pthread_create.c:456
#6  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 7 (Thread 0x7fb8a747f700 (LWP 24934)):
#0  0x00007fb8c3ab2acd in poll () at ../sysdeps/unix/syscall-template.S:84
#1  0x00005601d928299f in poll (__timeout=<optimized out>, __nfds=2, __fds=0x7fb8a747d360) at /usr/include/bits/poll2.h:46
#2  vo_x11_wait_events (vo=<optimized out>, until_time_us=<optimized out>) at ../video/out/x11_common.c:1899
#3  0x00005601d9267197 in wait_vo (until_time=1013374687, vo=0x5601db23bf00) at ../video/out/vo.c:682
#4  vo_thread (ptr=0x5601db23bf00) at ../video/out/vo.c:1006
#5  0x00007fb8c3d8636d in start_thread (arg=0x7fb8a747f700) at pthread_create.c:456
#6  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 6 (Thread 0x7fb8a7c87700 (LWP 24933)):
#0  0x00007fb8c3d9028d in read () at ../sysdeps/unix/syscall-template.S:84
#1  0x00005601d9224e8f in read (__nbytes=32768, __buf=0x7fb8a02226c0, __fd=<optimized out>) at /usr/include/bits/unistd.h:44
#2  fill_buffer (s=<optimized out>, buffer=0x7fb8a02226c0 "\373\311\301X\346$+\032\321d]\254\067T\326\003\201\334\020\377\201\325|\200'\245\300", max_len=32768) at ../stream/stream_file.c:81
#3  0x00005601d9222c7b in stream_read_unbuffered (s=s@entry=0x7fb8a0000a80, buf=0x7fb8a02226c0, len=len@entry=32768) at ../stream/stream.c:376
#4  0x00005601d9223534 in stream_read_partial (s=s@entry=0x7fb8a0000a80, buf=buf@entry=0x7fb8a02226c0 "\373\311\301X\346$+\032\321d]\254\067T\326\003\201\334\020\377\201\325|\200'\245\300", buf_size=buf_size@entry=32768) at ../stream/stream.c:424
#5  0x00005601d91cbe4e in mp_read (opaque=0x7fb8a0205810, buf=0x7fb8a02226c0 "\373\311\301X\346$+\032\321d]\254\067T\326\003\201\334\020\377\201\325|\200'\245\300", size=32768) at ../demux/demux_lavf.c:239
#6  0x00007fb8c91cfeeb in fill_buffer (s=0x7fb8a0209c00) at libavformat/aviobuf.c:566
#7  0x00007fb8c91d02ff in avio_seek (s=0x7fb8a0209c00, offset=1093744095, offset@entry=3, whence=<optimized out>, whence@entry=1) at libavformat/aviobuf.c:294
#8  0x00007fb8c91d0c52 in avio_skip (s=<optimized out>, offset=offset@entry=3) at libavformat/aviobuf.c:331
#9  0x00007fb8c91f5983 in flv_read_packet (s=0x7fb8a0208340, pkt=0x7fb8a7c85100) at libavformat/flvdec.c:954
#10 0x00007fb8c9311f3a in ff_read_packet (s=s@entry=0x7fb8a0208340, pkt=pkt@entry=0x7fb8a7c85100) at libavformat/utils.c:818
#11 0x00007fb8c9312bac in read_frame_internal (s=0x7fb8a0208340, pkt=0x7fb8a7c852e0) at libavformat/utils.c:1520
#12 0x00007fb8c9313a85 in av_read_frame (s=s@entry=0x7fb8a0208340, pkt=pkt@entry=0x7fb8a7c852e0) at libavformat/utils.c:1717
#13 0x00007fb8c9313ecb in seek_frame_generic (flags=0, timestamp=3136080, stream_index=0, s=0x7fb8a0208340) at libavformat/utils.c:2364
#14 seek_frame_internal (flags=0, timestamp=3136080, stream_index=0, s=0x7fb8a0208340) at libavformat/utils.c:2437
#15 av_seek_frame (s=0x7fb8a0208340, stream_index=stream_index@entry=-1, timestamp=timestamp@entry=3136080000, flags=flags@entry=0) at libavformat/utils.c:2457
#16 0x00005601d91cb98c in demux_seek_lavf (demuxer=0x7fb8a0205810, seek_pts=<optimized out>, flags=<optimized out>) at ../demux/demux_lavf.c:1039
#17 0x00005601d91c4787 in execute_seek (in=in@entry=0x7fb8a020e460) at ../demux/demux.c:845
#18 0x00005601d91c89e8 in demux_thread (pctx=0x7fb8a020e460) at ../demux/demux.c:869
#19 0x00007fb8c3d8636d in start_thread (arg=0x7fb8a7c87700) at pthread_create.c:456
#20 0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 5 (Thread 0x7fb8b49c7700 (LWP 24931)):
#0  0x00007fb8c3d8ceb6 in futex_abstimed_wait_cancelable (private=<optimized out>, abstime=0x7fb8b49c4d30, expected=0, futex_word=0x5601db232908) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
#1  __pthread_cond_wait_common (abstime=0x7fb8b49c4d30, mutex=0x5601db2328b8, cond=0x5601db2328e0) at pthread_cond_wait.c:539
#2  __pthread_cond_timedwait (cond=cond@entry=0x5601db2328e0, mutex=mutex@entry=0x5601db2328b8, abstime=abstime@entry=0x7fb8b49c4d30) at pthread_cond_wait.c:667
#3  0x00005601d91f8d33 in wait_wakeup (ctx=ctx@entry=0x5601db232810, end=end@entry=9223372036854775807) at ../player/client.c:314
#4  0x00005601d91fa36b in mpv_wait_event (ctx=0x5601db232810, timeout=1e+20) at ../player/client.c:784
#5  0x00005601d9213e81 in script_wait_event (L=0x7fb8ac000940) at ../player/lua.c:474
#6  0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8ac000940, func=func@entry=0x7fb8ac03c260, nresults=nresults@entry=1) at ldo.c:320
#7  0x00007fb8c597a792 in luaV_execute (L=L@entry=0x7fb8ac000940, nexeccalls=2, nexeccalls@entry=1) at lvm.c:591
#8  0x00007fb8c597001d in luaD_call (L=L@entry=0x7fb8ac000940, func=0x7fb8ac03c210, nResults=nResults@entry=0) at ldo.c:378
#9  0x00007fb8c596bcd5 in lua_call (L=L@entry=0x7fb8ac000940, nargs=nargs@entry=0, nresults=nresults@entry=0) at lapi.c:782
#10 0x00005601d9212f68 in load_scripts (L=0x7fb8ac000940) at ../player/lua.c:267
#11 0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8ac000940, func=func@entry=0x7fb8ac000d30, nresults=0) at ldo.c:320
#12 0x00007fb8c596ffd4 in luaD_call (L=0x7fb8ac000940, func=0x7fb8ac000d30, nResults=<optimized out>) at ldo.c:377
#13 0x00007fb8c596f31e in luaD_rawrunprotected (L=L@entry=0x7fb8ac000940, f=f@entry=0x7fb8c596a930 <f_call>, ud=ud@entry=0x7fb8b49c5160) at ldo.c:116
#14 0x00007fb8c597018b in luaD_pcall (L=L@entry=0x7fb8ac000940, func=func@entry=0x7fb8c596a930 <f_call>, u=u@entry=0x7fb8b49c5160, old_top=48, ef=<optimized out>) at ldo.c:464
#15 0x00007fb8c596bd68 in lua_pcall (L=L@entry=0x7fb8ac000940, nargs=nargs@entry=0, nresults=nresults@entry=0, errfunc=errfunc@entry=-2) at lapi.c:821
#16 0x00005601d9214ad2 in run_lua (L=0x7fb8ac000940) at ../player/lua.c:353
#17 0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8ac000940, func=func@entry=0x7fb8ac000d10, nresults=0) at ldo.c:320
#18 0x00007fb8c596ffd4 in luaD_call (L=0x7fb8ac000940, func=0x7fb8ac000d10, nResults=<optimized out>) at ldo.c:377
#19 0x00007fb8c596f31e in luaD_rawrunprotected (L=L@entry=0x7fb8ac000940, f=f@entry=0x7fb8c596a940 <f_Ccall>, ud=ud@entry=0x7fb8b49c53d0) at ldo.c:116
#20 0x00007fb8c597018b in luaD_pcall (L=L@entry=0x7fb8ac000940, func=func@entry=0x7fb8c596a940 <f_Ccall>, u=u@entry=0x7fb8b49c53d0, old_top=16, ef=ef@entry=0) at ldo.c:464
#21 0x00007fb8c596bdf7 in lua_cpcall (L=L@entry=0x7fb8ac000940, func=func@entry=0x5601d9214690 <run_lua>, ud=ud@entry=0x7fb8ac000900) at lapi.c:856
#22 0x00005601d9211bdd in load_lua (client=<optimized out>, fname=0x5601db238b00 "@stats.lua") at ../player/lua.c:386
#23 0x00005601d921a245 in script_thread (p=0x5601db2326f0) at ../player/scripting.c:95
#24 0x00007fb8c3d8636d in start_thread (arg=0x7fb8b49c7700) at pthread_create.c:456
#25 0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 4 (Thread 0x7fb8b51c8700 (LWP 24930)):
#0  0x00007fb8c3d8ceb6 in futex_abstimed_wait_cancelable (private=<optimized out>, abstime=0x7fb8b51c5d30, expected=0, futex_word=0x5601db22c37c) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
#1  __pthread_cond_wait_common (abstime=0x7fb8b51c5d30, mutex=0x5601db22c328, cond=0x5601db22c350) at pthread_cond_wait.c:539
#2  __pthread_cond_timedwait (cond=cond@entry=0x5601db22c350, mutex=mutex@entry=0x5601db22c328, abstime=abstime@entry=0x7fb8b51c5d30) at pthread_cond_wait.c:667
#3  0x00005601d91f8d33 in wait_wakeup (ctx=ctx@entry=0x5601db22c280, end=end@entry=9223372036854775807) at ../player/client.c:314
#4  0x00005601d91fa36b in mpv_wait_event (ctx=0x5601db22c280, timeout=1e+20) at ../player/client.c:784
#5  0x00005601d9213e81 in script_wait_event (L=0x7fb8a8000940) at ../player/lua.c:474
#6  0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8a8000940, func=func@entry=0x7fb8a8011300, nresults=nresults@entry=1) at ldo.c:320
#7  0x00007fb8c597a792 in luaV_execute (L=L@entry=0x7fb8a8000940, nexeccalls=2, nexeccalls@entry=1) at lvm.c:591
#8  0x00007fb8c597001d in luaD_call (L=L@entry=0x7fb8a8000940, func=0x7fb8a80112b0, nResults=nResults@entry=0) at ldo.c:378
#9  0x00007fb8c596bcd5 in lua_call (L=L@entry=0x7fb8a8000940, nargs=nargs@entry=0, nresults=nresults@entry=0) at lapi.c:782
#10 0x00005601d9212f68 in load_scripts (L=0x7fb8a8000940) at ../player/lua.c:267
#11 0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8a8000940, func=func@entry=0x7fb8a8000d30, nresults=0) at ldo.c:320
#12 0x00007fb8c596ffd4 in luaD_call (L=0x7fb8a8000940, func=0x7fb8a8000d30, nResults=<optimized out>) at ldo.c:377
#13 0x00007fb8c596f31e in luaD_rawrunprotected (L=L@entry=0x7fb8a8000940, f=f@entry=0x7fb8c596a930 <f_call>, ud=ud@entry=0x7fb8b51c6160) at ldo.c:116
#14 0x00007fb8c597018b in luaD_pcall (L=L@entry=0x7fb8a8000940, func=func@entry=0x7fb8c596a930 <f_call>, u=u@entry=0x7fb8b51c6160, old_top=48, ef=<optimized out>) at ldo.c:464
#15 0x00007fb8c596bd68 in lua_pcall (L=L@entry=0x7fb8a8000940, nargs=nargs@entry=0, nresults=nresults@entry=0, errfunc=errfunc@entry=-2) at lapi.c:821
#16 0x00005601d9214ad2 in run_lua (L=0x7fb8a8000940) at ../player/lua.c:353
#17 0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8a8000940, func=func@entry=0x7fb8a8000d10, nresults=0) at ldo.c:320
#18 0x00007fb8c596ffd4 in luaD_call (L=0x7fb8a8000940, func=0x7fb8a8000d10, nResults=<optimized out>) at ldo.c:377
#19 0x00007fb8c596f31e in luaD_rawrunprotected (L=L@entry=0x7fb8a8000940, f=f@entry=0x7fb8c596a940 <f_Ccall>, ud=ud@entry=0x7fb8b51c63d0) at ldo.c:116
#20 0x00007fb8c597018b in luaD_pcall (L=L@entry=0x7fb8a8000940, func=func@entry=0x7fb8c596a940 <f_Ccall>, u=u@entry=0x7fb8b51c63d0, old_top=16, ef=ef@entry=0) at ldo.c:464
#21 0x00007fb8c596bdf7 in lua_cpcall (L=L@entry=0x7fb8a8000940, func=func@entry=0x5601d9214690 <run_lua>, ud=ud@entry=0x7fb8a8000900) at lapi.c:856
#22 0x00005601d9211bdd in load_lua (client=<optimized out>, fname=0x5601db2324c0 "@ytdl_hook.lua") at ../player/lua.c:386
#23 0x00005601d921a245 in script_thread (p=0x5601db22c0f0) at ../player/scripting.c:95
#24 0x00007fb8c3d8636d in start_thread (arg=0x7fb8b51c8700) at pthread_create.c:456
#25 0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 3 (Thread 0x7fb8b59c9700 (LWP 24929)):
#0  0x00007fb8c3d8ceb6 in futex_abstimed_wait_cancelable (private=<optimized out>, abstime=0x7fb8b59c6d30, expected=0, futex_word=0x5601db225d6c) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
#1  __pthread_cond_wait_common (abstime=0x7fb8b59c6d30, mutex=0x5601db225d18, cond=0x5601db225d40) at pthread_cond_wait.c:539
#2  __pthread_cond_timedwait (cond=cond@entry=0x5601db225d40, mutex=mutex@entry=0x5601db225d18, abstime=abstime@entry=0x7fb8b59c6d30) at pthread_cond_wait.c:667
#3  0x00005601d91f8d33 in wait_wakeup (ctx=ctx@entry=0x5601db225c70, end=end@entry=9223372036854775807) at ../player/client.c:314
#4  0x00005601d91fa36b in mpv_wait_event (ctx=0x5601db225c70, timeout=1e+20) at ../player/client.c:784
#5  0x00005601d9213e81 in script_wait_event (L=0x7fb8b0000940) at ../player/lua.c:474
#6  0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8b0000940, func=func@entry=0x7fb8b00316a0, nresults=nresults@entry=1) at ldo.c:320
#7  0x00007fb8c597a792 in luaV_execute (L=L@entry=0x7fb8b0000940, nexeccalls=2, nexeccalls@entry=1) at lvm.c:591
#8  0x00007fb8c597001d in luaD_call (L=L@entry=0x7fb8b0000940, func=0x7fb8b00112b0, nResults=nResults@entry=0) at ldo.c:378
#9  0x00007fb8c596bcd5 in lua_call (L=L@entry=0x7fb8b0000940, nargs=nargs@entry=0, nresults=nresults@entry=0) at lapi.c:782
#10 0x00005601d9212f68 in load_scripts (L=0x7fb8b0000940) at ../player/lua.c:267
#11 0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8b0000940, func=func@entry=0x7fb8b0000d30, nresults=0) at ldo.c:320
#12 0x00007fb8c596ffd4 in luaD_call (L=0x7fb8b0000940, func=0x7fb8b0000d30, nResults=<optimized out>) at ldo.c:377
#13 0x00007fb8c596f31e in luaD_rawrunprotected (L=L@entry=0x7fb8b0000940, f=f@entry=0x7fb8c596a930 <f_call>, ud=ud@entry=0x7fb8b59c7160) at ldo.c:116
#14 0x00007fb8c597018b in luaD_pcall (L=L@entry=0x7fb8b0000940, func=func@entry=0x7fb8c596a930 <f_call>, u=u@entry=0x7fb8b59c7160, old_top=48, ef=<optimized out>) at ldo.c:464
#15 0x00007fb8c596bd68 in lua_pcall (L=L@entry=0x7fb8b0000940, nargs=nargs@entry=0, nresults=nresults@entry=0, errfunc=errfunc@entry=-2) at lapi.c:821
#16 0x00005601d9214ad2 in run_lua (L=0x7fb8b0000940) at ../player/lua.c:353
#17 0x00007fb8c596fb70 in luaD_precall (L=L@entry=0x7fb8b0000940, func=func@entry=0x7fb8b0000d10, nresults=0) at ldo.c:320
#18 0x00007fb8c596ffd4 in luaD_call (L=0x7fb8b0000940, func=0x7fb8b0000d10, nResults=<optimized out>) at ldo.c:377
#19 0x00007fb8c596f31e in luaD_rawrunprotected (L=L@entry=0x7fb8b0000940, f=f@entry=0x7fb8c596a940 <f_Ccall>, ud=ud@entry=0x7fb8b59c73d0) at ldo.c:116
#20 0x00007fb8c597018b in luaD_pcall (L=L@entry=0x7fb8b0000940, func=func@entry=0x7fb8c596a940 <f_Ccall>, u=u@entry=0x7fb8b59c73d0, old_top=16, ef=ef@entry=0) at ldo.c:464
#21 0x00007fb8c596bdf7 in lua_cpcall (L=L@entry=0x7fb8b0000940, func=func@entry=0x5601d9214690 <run_lua>, ud=ud@entry=0x7fb8b0000900) at lapi.c:856
#22 0x00005601d9211bdd in load_lua (client=<optimized out>, fname=0x5601db22be00 "@osc.lua") at ../player/lua.c:386
#23 0x00005601d921a245 in script_thread (p=0x5601db225850) at ../player/scripting.c:95
#24 0x00007fb8c3d8636d in start_thread (arg=0x7fb8b59c9700) at pthread_create.c:456
#25 0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 2 (Thread 0x7fb8b61ca700 (LWP 24928)):
#0  0x00007fb8c3ab2acd in poll () at ../sysdeps/unix/syscall-template.S:84
#1  0x00007fb8c3ad19fe in __poll_chk (fds=<optimized out>, nfds=<optimized out>, timeout=<optimized out>, fdslen=<optimized out>) at poll_chk.c:27
#2  0x00005601d9283f39 in poll (__timeout=-1, __nfds=<optimized out>, __fds=0x7fb8b61c8470) at /usr/include/bits/poll2.h:41
#3  terminal_thread (ptr=<optimized out>) at ../osdep/terminal-unix.c:399
#4  0x00007fb8c3d8636d in start_thread (arg=0x7fb8b61ca700) at pthread_create.c:456
#5  0x00007fb8c3abebbf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
Thread 1 (Thread 0x7fb8cbb1f180 (LWP 24927)):
#0  0x00007fb8c3d8ceb6 in futex_abstimed_wait_cancelable (private=<optimized out>, abstime=0x7ffdd3b7b8a0, expected=0, futex_word=0x5601db1ca3a0) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
#1  __pthread_cond_wait_common (abstime=0x7ffdd3b7b8a0, mutex=0x5601db1ca350, cond=0x5601db1ca378) at pthread_cond_wait.c:539
#2  __pthread_cond_timedwait (cond=cond@entry=0x5601db1ca378, mutex=mutex@entry=0x5601db1ca350, abstime=abstime@entry=0x7ffdd3b7b8a0) at pthread_cond_wait.c:667
#3  0x00005601d91e49e4 in mp_dispatch_queue_process (queue=0x5601db1ca340, timeout=0.24994299999999825) at ../misc/dispatch.c:311
#4  0x00005601d9216b5c in mp_wait_events (mpctx=0x5601db1c9e70) at ../player/playloop.c:64
#5  0x00005601d9218966 in run_playloop (mpctx=mpctx@entry=0x5601db1c9e70) at ../player/playloop.c:1113
#6  0x00005601d920eacd in play_current_file (mpctx=0x5601db1c9e70) at ../player/loadfile.c:1358
#7  mp_play_files (mpctx=mpctx@entry=0x5601db1c9e70) at ../player/loadfile.c:1519
#8  0x00005601d920f781 in mpv_main (argc=<optimized out>, argv=<optimized out>) at ../player/main.c:494
#9  0x00007fb8c39ce50a in __libc_start_main (main=0x5601d91a3ad0 <main>, argc=2, argv=0x7ffdd3b7bb48, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffdd3b7bb38) at ../csu/libc-start.c:295
#10 0x00005601d91a3b0a in _start ()
```

### Sample files
I wasn't playing this video, but this is first match for google search flv sample video.
http://www.mediacollege.com/video-gallery/testclips/barsandtone.flv

**Comment by Safari77** [comment]
also playing flac is broken.

**Comment by wm4** [comment]
It pretty much affects all IO through ffmpeg. They broke their API.

**Comment by myfreeer** [comment]
> @Safari77 also playing flac is broken. 

See #5017

**Comment by Safari77** [comment]
Looks like ffmpeg devs are happy with the situation.
So, is mpv getting a compatibility hack for both versions of EOF handling?

**Comment by wm4** [comment]
Just use Libav.

**Comment by sfan5** [comment]
upstream bug: https://trac.ffmpeg.org/ticket/6767
fix: #5033

**Comment by kevmitch** [comment]
seems fixed
